youtube_transcript_api/_api.py

```diff
@@ -99,47 +99,45 @@ def get_transcript(cls, video_id, languages=None, proxies=None):
 
 class _TranscriptFetcher():
     WATCH_URL = 'https://www.youtube.com/watch?v={video_id}'
-    API_BASE_URL = 'https://www.youtube.com/api/{api_url}'
-    LANGUAGE_REGEX = re.compile(r'(&lang=.*&)|(&lang=.*)')
+    API_BASE_URL = 'https://www.youtube.com/api/'
     TIMEDTEXT_STRING = 'timedtext?v='
+    NAME_REGEX = re.compile(r'(&name=.*&)|(&name=.*)')
 
     def __init__(self, video_id, languages, proxies):
         self.video_id = video_id
         self.languages = languages
+        print(languages)
         self.proxies = proxies
 
     def fetch(self):
         if self.proxies:
             fetched_site = requests.get(self.WATCH_URL.format(video_id=self.video_id), proxies=self.proxies).text
         else:
             fetched_site = requests.get(self.WATCH_URL.format(video_id=self.video_id)).text
-        timedtext_splits = fetched_site.split(self.TIMEDTEXT_STRING)
-        timedtext_url_start = (
-            timedtext_splits[2].find(self.TIMEDTEXT_STRING)
-            + len(timedtext_splits[0])
-            + len(timedtext_splits[1])
-            + len(self.TIMEDTEXT_STRING) + 1
-        )
-
-        for language in (self.languages if self.languages else [None,]):
-            response = self._execute_api_request(fetched_site, timedtext_url_start, language)
+        timedtext_splits = [split[:split.find('"')]
+                .replace('\\u0026', '&')
+                .replace('\\', '') 
+                for split in fetched_site.split(self.TIMEDTEXT_STRING)]
+        matched_splits = []
+        for language in self.languages:
+            matched_splits = [split for split in timedtext_splits if '&lang={}'.format(language) in split]
+            if matched_splits:
+                break
+        if matched_splits:
+            timedtext_url = min(matched_splits, key=self._sort_splits)
+            response = self._execute_api_request(timedtext_url)
             if response:
                 return response
 
         return None
 
-    def _execute_api_request(self, fetched_site, timedtext_url_start, language):
-        url = self.API_BASE_URL.format(
-            api_url=fetched_site[
-                timedtext_url_start:timedtext_url_start + fetched_site[timedtext_url_start:].find('"')
-            ].replace(
-                '\\u0026', '&'
-            ).replace(
-                '\\', ''
-            )
-        )
-        if language:
-            url = re.sub(self.LANGUAGE_REGEX, '&lang={language}&'.format(language=language), url)
+    #Sorting the matched splits by string length because we want non-asr options returned first
```

Comment:
Multiline docstrings for a method should be located below the method. Here's the PEP specification on this: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings